src/heap.py

```diff
@@ -12,28 +12,30 @@ def __init__(self, user_input=None):
             print('Argument is not iterable.')
 
     def _sift_up(self):
-        idx = len(self.heap) - 1
-        print('**************** sift up')
-        start = idx
-        while start > 0:
-            print('**** outer while')
-            idx = start
-            while idx > 0:
-                print('idx: ', idx, '  idx//2: ', idx//2)
-                print(self.heap[idx], '**', self.heap[idx//2])
-                if self.heap[idx] > self.heap[idx//2]:
-                    self.heap[idx], self.heap[idx//2] = self.heap[idx//2], self.heap[idx]
-                idx = idx//2
-            start = start//2
+        idx = self.length() - 1
+        while idx > 0:
+            child = idx
+            parent = (idx - 1) // 2
+            if self.heap[child] > self.heap[parent]:
+                self.heap[child], self.heap[parent] = self.heap[parent], self.heap[child]
+                idx = (idx - 1) // 2
+                continue
+            break
 
     def _sift_down(self):
         idx = 0
-        while idx < (len(self.heap) - 1)//2:
-            if self.heap[idx] < self.heap[idx*2] or self.heap[idx] < self.heap[(idx*2) + 1]:
-                bigger_child = max(self.heap[idx*2], self.heap[(idx*2) + 1])
-                child_idx = self.heap.index(bigger_child)
-                self.heap[idx], bigger_child = bigger_child, self.heap[idx]
-            idx = child_idx
+        while idx < (self.length() - 1) // 2:
+            print(idx)
+            parent = idx
+            child_l = idx * 2 + 1
+            child_r = idx * 2 + 2
+            idx += 1
+            print(self.heap)
+            if self.heap[parent] < self.heap[child_l]:
+                self.heap[parent], self.heap[child_l] = self.heap[child_l], self.heap[parent]
+            if self.heap[parent] < self.heap[child_r]:
+                self.heap[parent], self.heap[child_r] = self.heap[child_r], self.heap[parent]
```

Comment:
For this class, try and adhere to the PEP8 style guide and keep line lengths under 80.